File: src/main/java/org/dedda/bratwurst/lang/PrintChar.java
package org.dedda.bratwurst.lang;

import org.dedda.bratwurst.lang.scope.Scope;

/**
 * Created by dedda on 10/14/15.
 *
 * @author dedda
 */
public class PrintChar extends BWInstruction {

    private char toPrint;

    public PrintChar(int lineNumber, char toPrint) {
        super(lineNumber);
        this.toPrint = toPrint;
    }

    @Override
    public void run(Scope scope) {
        System.out.print(toPrint);
    }

    public char getToPrint() {
        return toPrint;
    }
}

File: src/main/java/org/dedda/bratwurst/lang/PrintChar.kt
package org.dedda.bratwurst.lang

import org.dedda.bratwurst.lang.scope.Scope

/**
 * Created by dedda on 10/14/15.
 *
 * @author dedda
 */
class PrintChar(lineNumber: Int, val toPrint: Char) : BWInstruction(lineNumber) {

    override fun run(scope: Scope) {
        print(toPrint)
    }
}